AssertionError: too many colons found - when trying to read from http links

**Describe the bug**
from the Error, It seems like an additional `https://` is being added to the provided link resulting in this error.

**To Reproduce**

```python
from daft import DataFrame
df = DataFrame.read_csv("https://github.com/DataTalksClub/nyc-tlc-data/releases/download/green/green_tripdata_2019-01.csv.gz")
```

same issue happens when trying to read from http s3 link (public data)

`df = DataFrame.read_csv("https://s3.amazonaws.com/nyc-tlc/misc/taxi+_zone_lookup.csv") # requires s3fs & boto`

**Expected behavior**
A daft dataframe.

**Desktop (please complete the following information):**
  Linux
  Python 3.8
  getdaft (0.0.22+dev0019.ccd06b4)

**Smartphone (please complete the following information):** N/A

**Additional context**
```
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
[<ipython-input-17-39d9e993993d>](https://localhost:8080/#) in <module>
      1 cs_path = "https://github.com/DataTalksClub/nyc-tlc-data/releases/download/green/green_tripdata_2019-01.csv.gz"
----> 2 df = DataFrame.read_csv(cs_path)

21 frames
[/usr/local/lib/python3.8/dist-packages/daft/filesystem.py](https://localhost:8080/#) in get_protocol_from_path(path, **kwargs)
     65 def get_protocol_from_path(path: str, **kwargs) -> str:
     66     split = path.split(":")
---> 67     assert len(split) <= 2, f"too many colons found in {path}"
     68     protocol = split[0] if len(split) == 2 else "file"
     69     return protocol

AssertionError: too many colons found in [https://https://github.com/DataTalksClub/nyc-tlc-data/releases/download/green/green_tripdata_2019-01.csv.gz](https://https//github.com/DataTalksClub/nyc-tlc-data/releases/download/green/green_tripdata_2019-01.csv.gz)
```

--- a/daft/filesystem.py
+++ b/daft/filesystem.py
@@ -2,6 +2,7 @@ from __future__ import annotations
 
 import dataclasses
 import sys
+from urllib.parse import urlparse
 
 if sys.version_info < (3, 8):
     from typing_extensions import Literal
@@ -80,9 +81,14 @@ def get_filesystem_from_path(path: str, **kwargs) -> AbstractFileSystem:
 ###
 
 
-def _fix_returned_path(protocol: str, returned_path: str) -> str:
+def _ensure_path_protocol(protocol: str, returned_path: str) -> str:
     """This function adds the protocol that fsspec strips from returned results"""
-    return returned_path if protocol == "file" else f"{protocol}://{returned_path}"
+    if protocol == "file":
+        return returned_path
+    parsed_scheme = urlparse(returned_path).scheme
+    if parsed_scheme == "" or parsed_scheme is None:
+        return f"{protocol}://{returned_path}"
+    return returned_path
 
 
 def _path_is_glob(path: str) -> bool:
@@ -99,7 +105,7 @@ def glob_path_with_stats(path: str) -> list[ListingInfo]:
     if _path_is_glob(path):
         globbed_data = fs.glob(path, detail=True)
         return [
-            ListingInfo(path=_fix_returned_path(protocol, path), size=details["size"], type=details["type"])
+            ListingInfo(path=_ensure_path_protocol(protocol, path), size=details["size"], type=details["type"])
             for path, details in globbed_data.items()
         ]
 
@@ -107,14 +113,14 @@ def glob_path_with_stats(path: str) -> list[ListingInfo]:
         file_info = fs.info(path)
         return [
             ListingInfo(
-                path=_fix_returned_path(protocol, file_info["name"]), size=file_info["size"], type=file_info["type"]
+                path=_ensure_path_protocol(protocol, file_info["name"]), size=file_info["size"], type=file_info["type"]
             )
         ]
     elif fs.isdir(path):
         files_info = fs.ls(path, detail=True)
         return [
             ListingInfo(
-                path=_fix_returned_path(protocol, file_info["name"]), size=file_info["size"], type=file_info["type"]
+                path=_ensure_path_protocol(protocol, file_info["name"]), size=file_info["size"], type=file_info["type"]
             )
             for file_info in files_info
         ]

--- a/tests/dataframe/test_creation.py
+++ b/tests/dataframe/test_creation.py
@@ -276,6 +276,13 @@ def test_create_dataframe_json_column_projection(valid_data: list[dict[str, floa
         assert len(pd_df) == len(valid_data)
 
 
+def test_create_dataframe_json_https() -> None:
+    df = DataFrame.read_json("https://github.com/Eventual-Inc/mnist-json/raw/master/mnist_handwritten_test.json.gz")
+    df.collect()
+    assert set(df.column_names) == {"label", "image"}
+    assert len(df) == 10000
+
+
 ###
 # Parquet tests
 ###

